src/Ssh/Ssh/Common/SshBaseCmdlet.cs

```diff
@@ -343,6 +373,48 @@ protected internal void ValidateParameters()
 
         }
 
+        protected internal void SetResourceType()
+        {
+            var resourcetypefilter = supportedResourceTypes.Select(type => $"resourceType eq '{type}'").ToArray();
+            String filter = $"$filter=name eq '{Name}' and ({String.Join(" or ", resourcetypefilter)})";
+            ODataQuery<GenericResourceFilter> query = new ODataQuery<GenericResourceFilter>(filter);
+
+            String[] types;
+            try
+            {
+                IPage<GenericResource> resources = ResourceManagementClient.Resources.ListByResourceGroupWithHttpMessagesAsync(ResourceGroupName, query).GetAwaiter().GetResult().Body;
+                types = resources.Select(resource => resource.Type).ToArray();
+            }
+            catch (CloudException exception)
+            {
+                throw new AzPSCloudException($"Failed to list resources in the {ResourceGroupName} Resource Group with error: \"{exception.Message}\". Ensure that the Resource Group Name is correct and that you have Read role on that resource group.");
```

Comment:
Suggest to move constant string to src/Ssh/Ssh/Properties/Resources.resx. See example [here](https://github.com/BethanyZhou/azure-powershell/blob/11b541bfb3c55a26c060fa274806d754f2dab8a0/src/KeyVault/KeyVault/Commands/ManagedHsm/UndoAzKeyVaultManagedHsmRemoval.cs#L99).

Reply:
I had no idea this existed. This is pretty cool.

Reply:
I moved the longer string constants to the resources. Didn't move all at this time because the change would be to large and harder to revert in case I later realize that moving the strings made development more difficult.